mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/SocketPolicy.java

```diff
@@ -56,17 +61,27 @@
   /**
    * Shutdown the socket input after sending the response. For testing bad
    * behavior.
+   *
+   * <p>Note: Be careful if this used in a test and it is not the final queued request. The client
+   * is free to continue as soon as it has received the response. If it makes a subsequent request
+   * the server may not have had time to closed the socket. Add delays in the client to improve
+   * the chances that the server has closed the socket.
    */
   SHUTDOWN_INPUT_AT_END,
 
   /**
    * Shutdown the socket output after sending the response. For testing bad
    * behavior.
+   *
+   * <p>Note: Be careful if this used in a test and it is not the final queued request. The client
```

Comment:
Not a fan of copy-pasting comments. This trains the reader to not read the documentation, which is bad. Promote this comment to top-level documentation?